rust/processor/src/processors/objects_processor.rs

```diff
@@ -0,0 +1,323 @@
+            PgPoolConnection,
+        },
+        util::standardize_address,
+    },
+};
+use anyhow::bail;
+use aptos_protos::transaction::v1::{write_set_change::Change, Transaction};
+use async_trait::async_trait;
+use diesel::{pg::upsert::excluded, result::Error, ExpressionMethods};
+use field_count::FieldCount;
+use std::{collections::HashMap, fmt::Debug};
+use tracing::error;
+
+pub struct ObjectsProcessor {
+    connection_pool: PgDbPool,
+}
+
+impl ObjectsProcessor {
+    pub fn new(connection_pool: PgDbPool) -> Self {
+        Self { connection_pool }
+    }
+}
+
+impl Debug for ObjectsProcessor {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        let state = &self.connection_pool.state();
+        write!(
+            f,
+            "ObjectsProcessor {{ connections: {:?}  idle_connections: {:?} }}",
+            state.connections, state.idle_connections
+        )
+    }
+}
+
+async fn insert_to_db_impl(
+    conn: &mut MyDbConnection,
+    (objects, current_objects): (&[Object], &[CurrentObject]),
+) -> Result<(), diesel::result::Error> {
+    insert_objects(conn, objects).await?;
+    insert_current_objects(conn, current_objects).await?;
+    Ok(())
+}
+
+async fn insert_to_db(
+    conn: &mut PgPoolConnection<'_>,
+    name: &'static str,
+    start_version: u64,
+    end_version: u64,
+    (objects, current_objects): (Vec<Object>, Vec<CurrentObject>),
+) -> Result<(), diesel::result::Error> {
+    tracing::trace!(
+        name = name,
+        start_version = start_version,
+        end_version = end_version,
+        "Inserting to db",
+    );
+    match conn
+        .build_transaction()
+        .read_write()
+        .run::<_, Error, _>(|pg_conn| {
+            Box::pin(insert_to_db_impl(pg_conn, (&objects, &current_objects)))
+        })
+        .await
+    {
+        Ok(_) => Ok(()),
+        Err(_) => {
+            conn.build_transaction()
+                .read_write()
+                .run::<_, Error, _>(|pg_conn| {
+                    Box::pin(async move {
+                        let objects = clean_data_for_db(objects, true);
+                        let current_objects = clean_data_for_db(current_objects, true);
+                        insert_to_db_impl(pg_conn, (&objects, &current_objects)).await
+                    })
+                })
+                .await
+        },
+    }
+}
+
+async fn insert_objects(
+    conn: &mut MyDbConnection,
+    items_to_insert: &[Object],
+) -> Result<(), diesel::result::Error> {
+    use schema::objects::dsl::*;
+    let chunks = get_chunks(items_to_insert.len(), Object::field_count());
+    for (start_ind, end_ind) in chunks {
+        execute_with_better_error(
+            conn,
+            diesel::insert_into(schema::objects::table)
+                .values(&items_to_insert[start_ind..end_ind])
+                .on_conflict((transaction_version, write_set_change_index))
```

Comment:
consider `do_nothing` for a transaction + event_indexer, this insertion is unlikely to change.

Reply:
for backfill i need `do_update` but after backfill we should change it back to `do_nothing`

Reply:
Yes but you don't need to change all the fields. Just the ones that you're backfilling.

Reply:
Oh actually you also needt o change inserted_at time otherwise it doesn't update.